lib/puppet-lint/data.rb

```diff
@@ -155,6 +155,21 @@ def title_tokens
       end
     end
 
+    # Internal: Determine if the given token contains a CLASSREF in
+    # the token chain..
+    #
+    # Returns a Boolean.
+    def classref?(token)
+      current_token = token
+      while true # rubocop:disable Style/InfiniteLoop
+        current_token = current_token.prev_code_token
+        break if current_token.nil?
```

Comment:
I think you can also write this as follows:
```suggestion
      while (current_token = current_token.prev_code_token)
```

I'm fine either way, but you avoid the RuboCop disable statement

Reply:
ohh nice

Reply:
TIL.. thanks @ekohl

Reply:
I think you removed one too many lines: the `current_token = token` line should be there, or you should use `token` all the time. RuboCop fails (correctly) on that now.

Reply:
yeah.. sharp eye... I got a bit happy with the `d` key 😄